ruby/examples/lib/register_user.rb

```diff
@@ -0,0 +1,22 @@
+class RegisterUser
+  Request = Struct.new(:email, :name, :password)
+  Response = Struct.new(:values, :errors)
+
+  attr_reader :gateway
+
+  def initialize(gateway: Gateway.new)
+    @gateway = gateway
+  end
+
+  def register(request)
+    errors = gateway.create_user(request.to_h)
+    Response.new(request.to_h, errors)
```

Comment:
Not sure why the response needs to contain the request values. Surely the caller knows what it sent?